src/Button/Button.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { fontSize, space } from 'styled-system'
+import { theme } from '../'
+
+const size = props => {
+  switch(props.size) {
+    case 'small':
+      return { 
+        height: '32px',
+        fontSize: `${theme.fontSizes[0]}px`,
+        padding: '0 12px'
+      }
+      break
+
+    case 'medium':
+      return {
+        height: '40px',
+        fontSize: `${theme.fontSizes[1]}px`,
+        padding: '0 18px'
+      }
+      break
+
+    case 'large':
+      return {
+        height: '48px',
+        fontSize: `${theme.fontSizes[2]}px`,
+        padding: '0 22px'
+      }
+      break
+
+    default:
+      return {
+        height: '40px',
+        fontSize: `${theme.fontSizes[1]}px`,
+        padding: '0 18px'
+      }
+      break
+  }
+}
+
+const Button = styled.button`
+  display: inline-block;
+  vertical-align: middle;
+  text-align: center;
+  outline: none;
+  text-decoration: none;
+  font-weight: 600;
+  cursor: pointer;
+  border-radius: ${props => props.radius ? props.radius : '2px'};
+  background-color: ${props => props.theme.colors.blue};
+  color: ${props => props.theme.colors.white};
+  border-width: 0;
+  border-style: solid;
+  
+  &:hover {
+    background-color: ${props => props.theme.colors.blue1};
+  }
+
+  &:disabled {
+    opacity: 0.25;
+  }
+
+  width: ${props => props.fullWidth ? '100%' : 'initial' };
+
+  ${size}
```

Comment:
Did you mean to add the `space` function here as well?

Reply:
I didn't try it, does it add `p`, `m`, `w` ?

Reply:
Yes, the `space` function adds these props (not `width` though): https://github.com/jxnblk/styled-system#space